jetty-servlet/src/test/java/org/eclipse/jetty/servlet/StatisticsServletTest.java

```diff
@@ -0,0 +1,217 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2016 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.servlet;
+
+import org.eclipse.jetty.http.HttpTester;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.server.LocalConnector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.ContextHandlerCollection;
+import org.eclipse.jetty.server.handler.StatisticsHandler;
+import org.eclipse.jetty.server.session.SessionHandler;
+import org.eclipse.jetty.xml.XmlParser;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.nio.ByteBuffer;
+
+/**
+ *
+ */
+public class StatisticsServletTest
+{
+    private Server _server;
+
+    private LocalConnector _connector;
+
+    @Before
+    public void createServer()
+    {
+        _server = new Server();
+
+        _connector = new LocalConnector( _server );
+        _server.addConnector( _connector );
+
+    }
+
+
+    @After
+    public void destroyServer()
+        throws Exception
+    {
+        _server.stop();
+        _server.join();
+    }
+
+    @Test
+    public void get_stats()
+        throws Exception
+    {
+        ContextHandlerCollection contexts = new ContextHandlerCollection();
+        _server.setHandler( contexts );
+
+        StatisticsHandler statsHandler = new StatisticsHandler();
+
+        ServletContextHandler statsContext = new ServletContextHandler( _server, "/" );
+        statsContext.setHandler( statsHandler );
+
+        _server.setHandler( statsContext );
```

Comment:
The handler wiring is a bit weird here.
You want the `StatisticsHandler` to be the outer most one, so it's:

```
Server -> StatisticsHandler -> ServletContextHandler
```

You don't need a `ContextHandlerCollection`.